app/domain/authentication/handler/authentication_handler.rb

```diff
@@ -130,7 +130,14 @@ def validate_client_ip(client_ip_address, conjur_role)
       end
 
       def fetch_conjur_role(authenticator, identity)
-        return @role_repository_class.from_username(authenticator.account, identity)
+        annotation = Annotation.fetch_annotation(
+          account: authenticator.account,
+          value: identity,
+          name: authenticator.authenticator_name
+        ).first
+        return @role_repository_class.from_username(authenticator.account, identity) unless annotation
+
+        @role_repository_class[role_id: annotation.resource_id]
```

Comment:
How come the default/base implementation for fetching a role is using annotations? I would expect this to be solely based on the ID, and I'm not sure why this change is necessary

Reply:
I agree with the order of selection concern. Let's use the following for the resolution order:

1. Find user by ID - essentially, this requires users to be in the root
2. Find user by annotation (`authn-oidc/identity`) - allows users to exist outside the root policy, but runs the risk of two users with the same OIDC identity

Defaulting to User ID as the default lookup maintains a similar behavior as other authenticators, but still allows for annotation based lookup.